### PR TITLE
AB#42: User Invitation Modal

### PR DESCRIPTION
**Pull request by kpjVideo**
This branch adds a few things:
- Error message gets displayed in the toast notification when org update fails (i.e. blank name, org already exists)
- Redirect when attempting to view /edit page of an org user is not in (getOrganizationUsers returns unauthorized)
- Invite user button and modal with form that validates email format

### CODE REVIEW

File: src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React, { useEffect, useState } from 'react';
 import NoOrganization from '../components/NoOrganization';
-import userService, { UserOrganization } from '../services/user.service';
```
Comment (zlix12) [inline src/pages/Dashboard.tsx:3]: It seems that we have an Organization type in `user.service` and in `organization.service` can we reduce this to just having the type in `organization.service`?

File: src/pages/EditOrganization.tsx [hunk 2]
```diff
@@ -33,22 +46,29 @@ const OrganizationForm = () => {
   const onSubmitForm = ({ name }: { name: string }) => {
     if (organization)
       OrganizationService.updateOrganization(organization?.id, { name })
-        .then((org) => {
-          setOrganization(org);
```
Comment (zlix12) [inline src/pages/EditOrganization.tsx:37]: A cleaner way may be `setOrganization({...organization, name});

**Review by zlix12 — CHANGES_REQUESTED** [review]
I get the edit organization error. Can we check `response.headers.get('Content-Type')` and if it is not `application/json` then do not attempt to process `response.json()`?

File: src/pages/Dashboard.tsx (hunk 1 above)
Comment (kpjVideo) [reply src/pages/Dashboard.tsx:3]: I think having organization.service import User from user.service and user.service importing Organization causes 'Dependency cycle' not sure if there's a way around that.

Comment (kpjVideo) [reply src/pages/Dashboard.tsx:3]: Actually at that point I think it makes sense to move those organization functions out of user service and into organization service.

Comment (zlix12) [reply src/pages/Dashboard.tsx:3]: @kpjVideo I agree with that.

**Review by zlix12 — APPROVED** [review]

**Pull request merged**